### PR TITLE
Feature/164/Add documents

### PR DESCRIPTION
**Pull request by tarasSanotskyi**
- [x] Create add documents step
- [x] Create general file uploader component
- [x] Add unit tests
- [x] Add file uploader to storybook
image
image
image

### CODE REVIEW

File: tests/unit/components/file-uploader/FileUploader.spec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  it('should render button with upload icon', () => {
+    const button = screen.getByText('test')
+    const icon = screen.getByTestId('CloudUploadIcon')
+
+    expect(icon).toBeInTheDocument()
+    expect(button).toBeInTheDocument()
+  })
+    
+  // it('should upload file', async () => {
```
Comment (maarekua) [inline tests/unit/components/file-uploader/FileUploader.spec.js:20]: Add a description with @TODO, or remove the comment.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: tests/unit/components/file-uploader/FileUploader.spec.js (hunk 1 above)
Comment (tarasSanotskyi) [reply tests/unit/components/file-uploader/FileUploader.spec.js]: fixed

File: src/components/file-uploader/file-uploader.style.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-around',
+    alignItems: 'center',
+    minHeight: '100px',
+    border: 'dashed',
+    borderColor: 'primary.200',
+    borderRadius: '4px',
+    overflow: 'auto',
+  },
```
Comment (rostyslaw27) [inline src/components/file-uploader/file-uploader.style.js:12]: don't duplicate styles
you can do
sx={ [styles.root, isDrag && styles.rootDrag] }
styles.rootDrag will override needed root styles

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/components/file-uploader/FileUploader.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import CloudUploadIcon from '@mui/icons-material/CloudUpload'
+import CloseIcon from '@mui/icons-material/Close'
+
+import { style } from '~/components/file-uploader/file-uploader.style'
+
+
+const FileUploader = ({ buttonText, upload }) => {
+  const { t } = useTranslation()
+
+  const { dragStart, dragLeave, dragDrop, addFiles, deleteFile, files, isDrag, error } = upload
```
Comment (maarekua) [inline src/components/file-uploader/FileUploader.js:13]: I think you should use a hook in this component and emit changes with a callback outside.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/components/file-uploader/file-uploader.style.js (hunk 2 above)
Comment (tarasSanotskyi) [reply src/components/file-uploader/file-uploader.style.js:12]: fixed

File: src/components/file-uploader/FileUploader.js (hunk 3 above)
Comment (tarasSanotskyi) [reply src/components/file-uploader/FileUploader.js]: fixed

File: src/constants/validation/files.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export const certificates = (files) => {
```
Comment (maarekua) [inline src/constants/validation/files.js:1]: This is not a constant. It is a helper (utils). And it should be general just receiving file size in MBytes.

File: src/containers/mentor-home-page/add-documents/AddDocuments.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import { useTranslation } from 'react-i18next'
+import { Box, Typography } from '@mui/material'
+
+import FileUploader from '~/components/file-uploader/FileUploader'
+import img from '~/assets/img/mentor-home-page/become-tutor/add-documents.png'
```
Comment (maarekua) [inline src/containers/mentor-home-page/add-documents/AddDocuments.js:5]: Can you name it addDocumentsPNG.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/containers/mentor-home-page/add-documents/AddDocuments.js (hunk 5 above)
Comment (tarasSanotskyi) [reply src/containers/mentor-home-page/add-documents/AddDocuments.js]: ok

File: src/constants/validation/files.js (hunk 4 above)
Comment (tarasSanotskyi) [reply src/constants/validation/files.js]: ok

**Review by maarekua — APPROVED** [review]

**Pull request merged**